From 0baecffed09f7e7135e3074651f0ca1d0192422d Mon Sep 17 00:00:00 2001
From: Andrey Albershteyn <andrey.albershteyn@gmail.com>
Date: Wed, 27 Sep 2023 10:30:58 +0200
Subject: [PATCH 2/2] common: fix linked binaries such as ls and true

The ls and true binaries on NixOS are located at different places.
Use 'type -P' and 'realpath' to find out where. Moreover, ls is
actually link too coreutils binary which behave based on file name.
So, make it behave like an 'ls' we need to rename it to 'ls'.
---
 tests/generic/452     | 4 ++--
 tests/generic/452.out | 4 ++--
 tests/xfs/031         | 3 ++-
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/tests/generic/452 b/tests/generic/452
index 6e14a1c4..5e62feb2 100755
--- a/tests/generic/452
+++ b/tests/generic/452
@@ -26,8 +26,8 @@ _exclude_scratch_mount_option "noexec"
 _scratch_mkfs > $seqres.full 2>&1
 _scratch_mount >> $seqres.full 2>&1
 
-LS="$(type -P ls)"
-SCRATCH_LS=$SCRATCH_MNT/ls_on_scratch
+LS="$(realpath $(type -P ls))"
+SCRATCH_LS=$SCRATCH_MNT/ls
 cp $LS $SCRATCH_LS
 
 $SCRATCH_LS $SCRATCH_LS | _filter_scratch
diff --git a/tests/generic/452.out b/tests/generic/452.out
index 1d24d6b3..436a01e1 100644
--- a/tests/generic/452.out
+++ b/tests/generic/452.out
@@ -1,3 +1,3 @@
 QA output created by 452
-SCRATCH_MNT/ls_on_scratch
-SCRATCH_MNT/ls_on_scratch
+SCRATCH_MNT/ls
+SCRATCH_MNT/ls
diff --git a/tests/xfs/031 b/tests/xfs/031
index 6e3813da..2175f3cc 100755
--- a/tests/xfs/031
+++ b/tests/xfs/031
@@ -56,11 +56,12 @@ lost+found d--755 3 1
 $
 EOF
 
+	true_bin=$(type -P true)
 	while [ $count -lt $total ]
 	do
 		let count=$count+1
 		cat >>$tmp.proto <<EOF
-${count}_of_${total}_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx ---755 3 1 /bin/true
+${count}_of_${total}_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx ---755 3 1 $true_bin
 EOF
 	done
 	echo '$' >>$tmp.proto
-- 
2.40.1

